test/python/circuit/test_unitary.py

```diff
@@ -149,6 +149,30 @@ def test_3q_unitary(self):
             self.assertIn(qubit.index, [0, 1, 3])
         assert_allclose(dnode.op.to_matrix(), matrix)
 
+    def test_1q_unitary_int_qargs(self):
+        """test single qubit unitary matrix with 'int' and 'list of ints' qubits argument"""
+        matrix = numpy.array([[1, 0], [0, 1]])
+
+        for num_q, qubits in enumerate([0, [0, 1]], 1):
```

Comment:
Wouldn't it be easier (both to code and understand 😉 ) to just check for the expected circuit?

Reply:
The test is similar to other tests in `TestUnitaryCircuit` class. Maybe the loop is a bit confusing (just to avoid repeating the same code for 1 and 2 qubits circuits). 
Are you suggesting that the following check would be more appropriate?
```suggestion
        from qiskit.extensions import UnitaryGate

X = np.array([[0, 1], 
              [1, 0]])

Z = np.array([[1, 0], 
              [0, -1]])

qr = QuantumRegister(2)

# new syntax
qc = QuantumCircuit(qr)
qc.unitary(X, 0)
qc.unitary(Z, [0, 1])

# expected circuit - old syntax
qc_target = QuantumCircuit(qr)
qc_target.append(UnitaryGate(X), [qr[0]])
qc_target.append(UnitaryGate(Z), [qr])

self.assertEqual(qc, qc_target)
```

Reply:
yes something like that and then compare if the circuits are equal, `self.assertEqual(qc_target, qc)`

Reply:
A new test was added. Please review the changes and let me know if anything is missing.